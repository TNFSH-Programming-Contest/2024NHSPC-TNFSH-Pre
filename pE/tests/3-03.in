2699
625726 920729 229773 660708 930818 577275 594463 176128 907497 584243 451182 797696 893588 610471 901381 478897 386331 395229 480626 723211 888101 770940 969516 655986 706907 408282 877679 801849 195783 193482 30254 231418 502651 981382 325381 755342 792557 873135 510209 958513 141862 376198 282768 508355 331291 817032 825430 187233 293835 289462 230560 513448 724254 741560 890822 273066 420006 514859 782412 859943 748477 503529 533401 339760 688996 88209 465127 441992 326167 497218 510854 204711 99793 948386 49191 469066 143307 190909 679182 465310 592808 382102 791994 588235 535545 14298 169386 966863 71463 497093 74637 142659 282766 248172 181829 715031 291046 824035 455778 967168 378584 321540 191072 782341 66330 341178 201371 433673 39380 829346 268036 192661 981648 488217 943160 357391 257083 135389 318455 400168 770329 337025 54975 478348 146620 813595 664071 450524 455429 405775 12958 703680 449176 315954 811782 817556 550333 402702 648894 745633 182098 437250 884386 116911 744790 88949 413427 162579 895798 952842 144263 86761 862377 530513 394725 323386 20853 804002 256929 109888 611364 165184 586138 202938 797129 573777 981147 206977 643342 422550 196746 946912 796018 719590 426957 750393 381221 938965 87075 670365 679042 402512 885489 834807 33848 117310 152095 495398 162158 956055 189091 433059 302580 784835 250677 453824 695702 766115 970345 567274 959141 731147 81103 538174 503056 394681 614773 593472 56994 178432 603428 730760 99202 448598 114241 162791 483093 710075 637714 499728 595457 55527 333598 652882 321305 986094 769874 265241 488058 919402 394052 39798 971063 880077 457622 193157 592483 528429 225707 368047 645060 959782 483498 973845 859779 524046 546274 24153 384574 87746 466929 630572 369630 932160 720667 76271 562967 704492 370902 331349 356993 319811 227180 34254 754505 560579 991051 786769 942926 185160 386530 158177 523429 553591 687423 661189 55996 298533 799556 733295 405124 469249 733336 482224 731523 339279 336262 226998 303924 213255 826346 210533 987375 20534 575694 13752 282576 277086 399615 325974 82098 795697 682254 464360 288134 798387 247718 462811 593057 559914 606731 71524 998546 905386 953936 653532 283590 987866 970559 700508 111034 44074 249623 57265 565092 50889 351089 416754 504450 758073 269327 766526 342279 913152 15905 922943 101397 572101 313412 128171 362927 630983 505354 24808 829355 169252 340927 171906 310324 455750 477733 62099 346426 514031 806250 277495 790614 42903 129305 692385 945616 262812 657757 200612 858569 332910 195795 883706 770233 851870 184513 406959 475541 568929 402868 383090 977992 242814 657480 844306 757916 115578 224346 939564 547067 250701 154741 918071 394737 188939 354156 378289 577006 484304 910075 568808 535152 464855 229810 245276 87496 858621 838404 355469 123159 652523 353877 515276 528211 168018 311431 56989 815365 954620 844515 126319 994500 309007 91789 703455 524488 591501 106845 47679 956306 679087 204488 173566 581205 436313 350504 125312 855441 168435 263446 42619 379312 61665 363222 295423 144732 199897 278193 816237 553890 72923 717953 584670 666488 815535 452133 738610 557902 107026 162360 236848 680377 347726 555233 646799 790717 801062 359914 441322 393797 318268 772762 726917 223496 572573 302545 104578 533493 810194 950794 303401 304473 496668 311579 731451 577629 644338 838679 670578 429500 630749 367770 49873 409431 275137 22443 91115 350926 226617 747262 449269 754915 694879 813490 956145 624826 114805 345463 411334 976611 357168 660527 812942 487517 951196 603641 516330 417432 290396 824718 702772 879597 321784 132009 179768 311115 476916 880980 487497 691964 176032 710289 946390 759485 425149 360997 195370 816987 370361 597170 988438 894197 44264 695085 949561 63188 446375 284021 555816 613969 915495 905534 206547 238724 797465 238699 389549 686855 264698 675411 814788 443553 475873 163657 36948 332110 905981 704231 918180 188225 919488 218306 284132 454189 84349 941805 533533 411505 513876 325316 306493 893784 80291 888699 824799 217893 117142 40097 723727 233336 284055 296249 898675 631528 736715 521066 222251 247910 902786 453231 451925 593446 692828 839874 639204 365558 797689 89635 576291 223896 785498 178101 913311 732329 142467 147083 761183 314067 944768 393751 650245 458493 88635 41020 186491 604953 92507 787473 685279 163472 315114 522569 93820 992892 397308 717522 894318 790660 199658 201778 811307 772238 268711 306012 697415 594741 235831 782098 717804 190426 730866 688283 474577 800382 766929 242656 547838 165607 14561 718002 785401 175286 242463 462337 174183 732587 519507 568826 261333 19139 984812 464253 459340 707821 322802 296038 81576 350189 309429 737485 833044 422967 813561 45854 885060 53781 594673 402144 651653 48418 446549 848503 83695 276351 337635 984891 16713 332239 197015 294180 159672 402892 593634 369678 911833 167767 317983 887012 179493 299085 394554 489181 431249 636063 461150 39869 504122 573849 459007 824134 231264 713966 470410 538646 112365 29470 753152 889153 518335 63674 111467 644765 340343 992919 497909 943484 692165 195992 219107 41641 733048 603433 328210 819967 730215 492820 130266 925877 133353 350459 101924 49844 500772 111543 172142 797514 722578 25518 163404 667684 626179 568490 85203 573359 439693 985426 22698 106996 138437 441900 558333 625769 742047 349083 864127 161508 76429 902017 59495 994660 628209 432121 440588 288938 325312 447661 565697 59913 275007 851434 396280 975437 357421 731088 240026 634214 572384 942854 239444 206088 950111 802632 776417 107110 607209 687995 253602 347268 405611 453162 954224 550535 918478 257185 287582 31727 882128 634486 372858 767199 839508 148389 928789 281874 379998 753029 99531 657442 430718 694570 777337 494699 80195 474201 931632 764423 836182 876667 632354 432369 780379 412993 926095 440438 15504 101758 554091 527134 217632 834791 994276 779999 178694 15359 954347 508833 136372 118757 910384 996270 852573 165661 923464 384349 186834 987232 51749 889551 360075 329088 505359 675724 980189 800362 500202 424574 493786 588919 838576 359856 321858 295959 681689 778877 435356 394155 178485 138313 566759 426309 913944 831256 651587 532131 672177 246709 618198 18862 153059 65978 972537 731087 851179 300513 538915 374567 600537 420801 276893 430274 185536 228585 865958 695800 134343 454364 816998 511335 597096 477988 311151 894163 281810 237202 790574 112563 20115 34415 869097 920660 537476 900798 482713 382498 251151 885265 714868 688404 379190 609461 288192 941272 315045 517091 817700 454287 959227 675184 517940 662211 573525 690730 11334 221333 245968 368062 513023 307415 149215 66424 310067 230440 875377 689083 946063 98993 329459 453465 465772 496674 781625 682717 328428 530231 842726 910959 38534 268690 621040 532275 139173 279264 559602 82910 361432 836156 897288 673155 811112 558845 530347 505087 483541 189575 700807 209506 846803 278330 593094 39418 797830 132801 602916 686946 808001 751388 434426 766059 458995 835306 442355 212422 516626 652334 580657 458431 715749 564964 300721 967214 572338 230797 395459 562720 87618 613768 458225 15333 526891 575313 923669 426560 645645 811253 227726 179479 50116 397149 576018 765813 808467 481809 648450 384619 396937 997646 374360 94785 53225 940191 377438 259716 944755 899951 577556 965903 188604 80355 674138 339328 862821 414743 778606 318787 401822 191288 216541 839418 270115 658271 80500 144879 396071 865069 809817 863346 951028 813382 743139 376121 819972 450088 600035 753373 894816 525022 397457 19428 365025 768394 547339 753802 451034 322809 479198 481123 404448 451205 913125 316458 366355 143413 210242 192678 177117 548543 617779 94426 805517 44626 173107 207120 545908 120479 846504 368948 181483 938076 931507 587222 626608 371351 452704 56122 966725 134388 968505 563049 224165 370955 176578 682504 583794 977668 577477 284476 654514 38100 421954 269246 539265 962330 694126 763208 391468 288835 99072 190250 909981 430509 772513 802899 15082 877797 91425 328765 255508 878692 468446 208640 63488 915353 507666 358655 315406 716200 396967 519765 150007 893755 687873 471659 322257 770600 599262 683034 245274 265401 273318 550095 549992 234971 766326 79974 177322 926363 486047 216218 681613 989447 619964 851649 973351 109640 403055 151165 613155 211289 613470 134664 828202 276706 962269 325599 926668 471666 283190 193200 287120 609458 519185 990952 95850 27529 779901 885132 903555 67645 391983 908848 336686 603924 34204 104758 29989 585483 53718 488704 26617 81464 379185 888821 932828 60384 450089 475291 927922 119691 84264 961614 746407 66272 264780 293495 835428 540015 918065 526083 406925 700038 865773 619114 527286 762508 870506 387631 509659 250708 830656 126994 409706 712363 412595 350757 571655 935227 421950 534334 668542 242434 53624 725786 784269 453274 775401 721015 953982 455675 935741 139346 927616 471021 630607 546049 203950 726171 657838 136851 799871 462952 983999 831656 378071 684209 71319 140004 778916 459371 553162 845051 772781 960103 587940 166522 98363 340045 525418 947527 471044 869954 255495 574209 661295 321256 698402 895734 40859 730386 296181 132763 823325 829747 708457 544048 527039 491290 995739 338261 219869 203784 772392 145046 491669 758408 660881 605533 751066 907283 46338 355906 221044 561873 826558 658671 980793 371364 406206 884200 309343 465117 508932 668804 678710 663559 454611 134547 107108 911175 197921 57449 445855 628174 327037 954204 732094 82896 532998 122950 80100 940356 25041 58707 832983 592998 611226 32709 359590 790092 681474 943739 790586 707258 569658 428329 952115 918917 487749 972927 315616 147655 583608 18963 937698 622977 373805 949953 974026 19307 530831 333021 271944 990692 686457 22650 630631 851244 576587 216144 14721 972487 367616 616502 758434 411672 446426 456919 960956 831194 665753 196133 308068 187104 139487 969015 79656 214850 127994 802076 297932 724342 847317 694234 644898 446475 331703 826907 30015 103330 141446 300503 21552 466327 725099 857948 399852 756923 274996 164070 717571 65124 550213 559993 472587 531426 830821 947805 501771 242392 392429 329809 720884 164189 223764 695185 540490 268380 904418 923588 643185 169452 949955 128371 899739 911082 482291 799990 663980 278063 119659 320536 870624 739238 957269 511066 475066 227526 819251 537828 95460 545023 979172 493784 170223 603569 529921 585754 458284 877745 496297 772821 977276 969641 584452 650555 700796 280159 813268 455662 243628 685319 796612 32720 727988 608285 734100 360776 40316 991403 181205 193393 155620 995492 981585 366166 823586 765455 515429 722358 16669 871784 798464 584275 196777 949925 969970 197085 342465 56735 556827 860199 588088 832095 347925 637411 79035 512200 729873 301195 942177 994142 797614 592829 718563 997878 406306 177380 622488 139782 329264 413995 744197 912193 310136 660501 586788 331116 85419 905953 372913 524362 987246 313515 425664 991484 817704 788870 554217 323749 969004 423341 142738 17560 211799 520187 652553 274287 158445 87115 598421 491092 57559 218018 520464 404365 476428 868105 632234 262546 499661 822626 158048 307395 826349 251974 985700 83408 330399 219729 763170 166084 608280 204004 183597 575869 485188 819451 853043 252327 798426 970796 784363 737225 369125 769056 342299 564174 670071 625349 127554 31447 265202 997483 486417 739776 229045 943226 732594 80427 153834 179399 982297 531880 573435 484655 785079 675831 286680 311274 205129 865796 187005 47975 520441 991312 162483 336171 537795 277527 194425 597074 891315 619050 904371 42025 228783 669113 689302 863625 568249 284267 256478 435310 31399 360544 218251 536006 111653 856677 834365 70670 476268 706244 673247 863501 128299 13692 927832 672218 50317 297506 518676 721595 51895 822827 490051 939450 293100 883733 631255 597676 334795 385992 570821 798353 547868 375350 787506 403710 673523 856537 629142 165671 846468 699939 59807 368256 897820 450954 293725 936833 56872 221049 489025 22755 985669 525628 357069 514843 491237 795180 131443 826498 497127 196861 564473 196918 475486 530814 947891 502824 137498 997051 846529 106485 268821 10191 486211 783285 36171 214102 676958 137863 761177 684430 634410 708390 720851 228116 372239 268461 812985 490758 717910 345525 34482 334080 177795 887545 911840 815490 80511 516910 677555 238659 128550 896507 323196 840329 435779 511135 653846 441685 219320 402200 115842 407086 187255 269332 476830 948756 104700 88452 708343 784986 209755 202195 655257 525913 596487 885917 47114 133778 73216 193887 70493 552524 488841 91477 167796 278984 636140 191558 715531 130490 583582 414596 40830 567432 990238 57883 696608 991121 171881 230123 913798 910915 114838 663122 25443 577691 183027 821774 427166 91591 945524 356604 718398 621305 178359 420317 95981 986882 765794 711022 720315 123717 189284 148485 482697 953825 651905 563773 456919 589106 150787 535279 547562 952898 715151 727615 109354 376763 179972 237218 15009 739899 486162 25654 768219 150457 641021 944727 624885 585525 346454 904925 214655 577842 393244 344119 889991 22372 204227 146452 588849 442753 325191 860434 590109 240851 904791 829982 312078 567098 48178 686398 996063 381000 405660 949860 526575 97889 770537 854968 696480 801809 192665 991643 805098 283307 688224 749468 574848 926345 398650 79574 169597 596070 265522 173562 198635 945734 572028 804260 652099 483810 717740 724684 175379 239382 894486 372305 485510 723950 378394 251379 747948 570883 769607 634910 710550 52002 291337 715490 589613 319361 207732 412602 204060 702597 705449 614300 206982 589798 70488 701481 563497 141237 743702 552799 43055 404937 926253 603510 602518 928133 714175 873034 95255 170234 618983 686913 766746 918563 177860 126682 374252 956800 221542 119105 929633 525319 506289 94073 792534 993289 832707 341969 409871 731901 641863 61983 910173 524527 871682 931215 170290 576017 302670 44207 894228 741515 632995 694148 477380 884805 198500 621824 566982 292137 519563 360630 910138 980744 645162 532235 425100 877391 674268 982251 77418 549626 966315 305611 870543 828846 230850 628063 390309 623366 166102 26072 252816 866971 409716 90505 882579 786326 865118 793754 203761 653722 357258 127258 187939 924642 172180 359052 64763 150064 284651 892404 967630 565631 498115 194085 27950 938944 687248 433569 610605 510285 894901 78134 954000 905267 600878 864627 855948 868770 940354 289587 385276 985999 625633 894962 80997 769435 787381 446101 72592 940028 41390 808091 908449 373307 929349 530891 440874 757178 999227 823101 720030 698703 314530 158962 194664 197829 544212 660367 902513 91209 845340 48841 579034 115670 244324 449804 30367 201583 759656 94739 392405 136327 612467 441913 857857 585087 306669 843501 82842 552686 151288 729061 192283 741941 628785 124670 668581 393021 822629 560054 965781 482062 76720 562870 369045 275788 579408 383817 807351 295672 250037 969367 375388 648162 516913 760662 936411 964173 443795 203507 349153 843041 33972 171183 419843 156782 925232 325909 545642 353354 415419 840367 51928 256645 672717 532028 610881 228358 715114 247120 15259 28233 120187 398332 309863 305659 516769 699189 520047 712107 761918 471957 949066 480332 614606 974839 919370 703309 375712 64346 538751 944182 650918 850538 643480 634138 534144 741637 586994 822250 864324 833629 216025 626440 40992 390933 620428 169181 772118 335052 920096 536769 998122 549753 426431 166987 965204 968028 166777 46805 933101 532093 562799 756043 778837 762051 233060 33487 71863 186880 751704 798841 552023 215941 802153 725142 633398 845632 28768 951054 117050 708303 204788 905933 248132 977510 54578 906471 920956 771253 496902 216713 912062 393057 119546 421186 758230 366420 555505 493491 344154 82601 811549 583588 938757 365665 927598 659569 604216 17223 62972 378880 49822 691517 995175 213616 716378 983342 875813 681794 565949 949468 205428 645325 373378 415500 918351 143299 962347 233842 90045 671756 852334 881066 475320 886976 625036 656220 837259 991316 49575 269312 189986 673176 253603 446107 17590 550251 383192 740465 457084 390563 164093 592812 560136 287782 698433 940595 574604 720470 875038 70971 492774 640515 512218 857230 981114 195819 241665 197328 555535 425393 202731 110350 366682 378508 246460 367220 105409 597637 128261 263325 985489 163413 956236 617878 48818 269461 438251 363696 544984 600482 29340 395050 949525 698459 686705 382344 225399 780915 749718 459305 357282 67089 808114 994473 492391 598704 218984 638312 264711 465198 331728 118680 260893 169735 841951 314522 467581 320581 771892 960720 71897 688990 230414 417257 779807 470203 545051 864863 558077 988076 772945 809022 504630 57619 617754 209258 542147 958330 259652 465183 386048 773028 634260 272346 268131 999853 262572 17312 648840 66899 759994 193490 292287 777164 989680 454303 508302 595423 142759 254487 615807 506138 667314 774115 784035 125687 306290 592742 459310 758598 871244 382453 847675 463183 403283 547745 86931 946502 113241 372073 319358 815439 735021 552216 30696 223866 470982 674750 34365 283974 987737 297591 813499 25734 295686 319635 909521 429058 126689 470666 493495 246755 617421 635775 506438 621342 755478 20388 868721 249776 557854 499595 441294 430759 804961 537361 990673 411735 693164 480073 790560 357378 606956 425881 188169 545639 101349 110701 728435 140970 370831 498351 927384 956873 540791 768574 696753 103318 172321 134009 901343 453568 906069 971920 688059 506334 35572 330630 368616 615089 614486 658808 614244 786720 210785 689470 494773 119985 660089 210361 753990 217979 910351 506295 945691 66350 914451 646350 693754 247410 765109 906943 369935 752074 80241 166395 643805 423338 652711 149111 292146 820718 58242 349885 383854 321870 806174 155448 901885 609911 190501 745641 30936 446469 917399 995267 205966 196615 766130 215933 855923 289430 939880 489474 670879 905114 551839 389015 799189 848990 462643 526264 986038 82470 734488 55109 395961 681401 902814 434177 584229 908782 632676 467227 384364 928989 512612 154515 436273 556168 819913 663524 460666 235504 312267 605966 702670 459770 183065 852805 992343 94369 798545 187089 823971 360233 172959 409980 806948 860479 334840 784292 708541 208263 273381 366126 909805 181392 89183 852330 511551 210137 488299 955953 795298 745963 499473 189424 269169 770381 215877 926039 358550 910776 647273 692614 669078 355815 492099 985908 167023 167146 576817 931611 801392 493633 484562 846565 947339 561962 765967 103304 732151 548979 67472 46985 236277 48652 271025 387966 464218 87852 676831 940197 317828 890324 874974 603053 748745 857401 524777 724172 527199 10122 703799 231088 525839 199074 121578 32479 953969 518971 982926 644286 626648 774034 570170 365704 890089 90934 356844 386438 470626 247128 310281 391042 625035 605768 543070 214981 480801
