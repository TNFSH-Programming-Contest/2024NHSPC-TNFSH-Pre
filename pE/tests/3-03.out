208576674433389